### PR TITLE
Extract env setup to a script

### PR DESCRIPTION
**Pull request by noamnelke**
Also, support different operating systems and architectures when installing protobuf.

### CODE REVIEW

File: setup_env.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+./ci/install-protobuf.sh
+
+go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
```
Comment (antonlerner) [inline setup_env.sh:9]: you are accessing github.com a lot, i'd put it in a pramaeter $GITHUB

File: ci/install-protobuf.sh [hunk 2]
```diff
@@ -1,12 +1,35 @@
...
+else
+    echo "unsupported OS, protoc not installed"
+    exit 1;
+fi
+protoc_url=https://github.com/protocolbuffers/protobuf/releases/download/v3.6.1/protoc-3.6.1-${os_arch}.zip
+
 # Make sure you grab the latest version
-curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.6.1/protoc-3.6.1-linux-x86_64.zip
+echo "fetching ${protoc_url}"
+curl -L -o "protoc.zip" ${protoc_url}
```
Comment (antonlerner) [inline ci/install-protobuf.sh:16]: what if this fails? maybe put in && or $? && cmd

Comment (noamnelke) [reply ci/install-protobuf.sh:16]: solved in 597e6a8

File: setup_env.sh (hunk 1 above)
Comment (noamnelke) [reply setup_env.sh:9]: I think the readability and code complexity tradeoff isn't worth the DRY-ness

File: ci/install-protobuf.sh [hunk 3]
```diff
@@ -1,12 +1,33 @@
...
+unzip -u protoc.zip -d protoc3
 
 # Move protoc to /usr/local/bin/
+echo "moving bin to /usr/local/"
 sudo mv protoc3/bin/* /usr/local/bin/
 
 # Move protoc3/include to /usr/local/include/
-sudo mv protoc3/include/* /usr/local/include/
+echo "syncing include to /usr/local/include"
+sudo rsync -a protoc3/include/ /usr/local/include/
```
Comment (antonlerner) [inline ci/install-protobuf.sh]: remove sudo

**Review by antonlerner — APPROVED** [review]

**Review by antonlerner — APPROVED** [review]

**Pull request merged**